src/components/Chart/index.js

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable no-new */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import ChartJS from 'chart.js';
+import { Provider } from './context';
+import resolveOptions from './resolve-options';
+import './styles.css';
+
+export default class Chart extends Component {
+    constructor(props) {
+        super(props);
+        this.registerDataset = this.registerDataset.bind(this);
+        this.chartRef = React.createRef();
+        this.datasets = [];
+    }
+
+    componentDidMount() {
+        this.renderChart();
+    }
+
+    registerDataset(datasetValues) {
+        this.datasets = this.datasets.concat(datasetValues);
+    }
+
+    renderChart() {
+        const {
+            type,
+            labels,
+            disableAnimation,
+            disableLines,
+            disableCurves,
+            showLegend,
+            legendPosition,
+            showStacked,
+        } = this.props;
+        const node = this.chartRef.current;
+        const datasets = this.datasets;
+
+        new ChartJS(node, {
+            type,
+            data: {
+                labels,
+                datasets,
+            },
+            options: resolveOptions(
+                disableAnimation,
+                disableLines,
+                disableCurves,
+                showLegend,
+                legendPosition,
+                showStacked,
+            ),
+        });
+    }
+
+    render() {
+        const { children } = this.props;
+        return (
+            <div>
+                <span className="rainbow-chart_dataset-container" aria-hidden>
+                    <Provider value={{ privateRegister: this.registerDataset }}>
```

Comment:
every time you do something like prop={{ value  }} ... the object gets create in every execution of the render method. You can cache this instead by doing 
```
this.contextValue = {
    privateRegister: this.registerDataset.bind(this),
}
```

then in the component render you pass the contextValue directly .. created only one time

Reply:
other observation, use Symbol to hide private methods and variables